### PR TITLE
fix: aligned TEI examples to specification

### LINKED ISSUE #87: BUG: Examples don't match specification
https://github.com/CycloneDX/transparency-exchange-api/blob/main/discovery/readme.md#tei-syntax

Examples don't match TEI specifcation.

### PR DESCRIPTION
**Pull request by madpah**
Resolves #87

### CODE REVIEW

**Comment by ppkarwasz** [comment]
Maybe the examples are right and the specification should be modified. The first element you might expect in a URN is the authority.

**Comment by madpah** [comment]
Fair comment @ppkarwasz - @oej - thoughts on the order of the items in TEI?

**Comment by oej** [comment]
The TYPE needs to come first as we can come up with TEI types that has another resolution than DNS in the future.

**Comment by ppkarwasz** [comment]
Aren't there two `<type>`s then?

- One tells us how to find the TEA service.
- One is part of the identifier

For example:

- `urn:tei:dns:logging.apache.org:purl:pkg:maven/org.apache.logging/log4j-core` tells the user to use the DNS discovery mechanism we were talking about.
- `urn:tei:purl:pkg:maven/org.apache.logging/log4j-core` tells the user to use an ecosystem-dependent mechanism to find the TEA server.

Alternatively, to keep it simple:

- The TEI identifier **only** refers to the DNS mechanism and we have: `urn:tei:logging.apache.org:purl:pkg:maven/org.apache.logging/log4j-core`.
- If a user doesn't find a TEI, but find the PURL `pkg:maven/org.apache.logging/log4j-core`, he can still use a TEA service, but the resolution mechanism is ecosystem-dependent.

**Comment by oej** [comment]
I don't like restricting the TEI to ONLY DNS resolution. There will be a lot of work establishing the TEI as a common identifier and having to restart that work for new non-DNS types will be hard. Better to have one single identifier and be able to extend it.

But you are not wrong that the PURL could be used for TEA discovery too, especially for stuff that is not packaged. That is a discussion we need to bring up in the PURL wg, I just haven't had time for it. We mentioned it briefly as an option in the SBOM forum vulnerabilities wg meeting. I'm thinking of commercial software and open source projects not primarily being distributed as packages (like C code).

**Comment by oej** [comment]
We really need to clarify the intent of the TEI. But that's another PR

**Comment by oej** [comment]
Created issue #89 to clarify the TEI URN more. It's definitely needed for registration with IANA/IETF

**Review by oej — APPROVED** [review]
Time to merge. Discussions are moved to other tickets. Thank you for this PR!

File: discovery/readme.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ urn:tei:purl:<domain-name>:<purl>
 
 Example:
 ```text
-urn:tei:org.cyclonedx:pkg:pypi/cyclonedx-python-lib@8.4.0?extension=whl&qualifier=py3-none-any
+urn:tei:purl:org.cyclonedx:pkg:pypi/cyclonedx-python-lib@8.4.0?extension=whl&qualifier=py3-none-any
```
Comment (oej) [inline discovery/readme.md:79]: The domain should be "cyclonedx.org"

File: discovery/readme.md [hunk 2]
```diff
@@ -120,7 +120,11 @@ Syntax:
 urn:tei:uuid:<domain-name>:<uuid>
 ````
 
-Has to be a valid UUID.
+Example:
+```text
+urn:tei:uuid:cyclonedx.org:d4d9f54a-abcf-11ee-ac79-1a52914d44b1
```
[the remark this answers is not in the document]
Comment (oej) [reply discovery/readme.md:125]: Resolved in other conversation. The UUID includes version.

**Review by oej — CHANGES_REQUESTED** [review]

**Comment by madpah** [comment]
Requested changes made @oej

**Review by oej — APPROVED** [review]
Great work. Thank you Paul!

**Pull request merged**